docs/src/proposals/append-vec-storage.md

```diff
@@ -0,0 +1,231 @@
+|                       | for rent epoch            |
++---------------------------------------------------+
+| footer                | 40 bytes                  |
++---------------------------------------------------+
+```
+
+### Better Compression for Account Data
+All account-data is put together into one block to make them easier
+to compress, compared to having them stored interleaving with the pubkeys.
+
+### Per-Account Hash is replaced by AppendVec Block Hash
+The per-account hash is removed.  Instead, each block in the new AppendVec file
+has one 32-bytes hash, and the hash for the entire AppendVec is also included
+in the footer.
+
+### Format Details
+
+#### Header
+
+```
++-----------------------------------------+
+| Header                                  |
++-----------------------------------------+
+| format version      | 8 bytes           |
+| account meta offset | 8 bytes           |
+| pubkey_offset       | 8 bytes           |
+| owners_offset       | 8 bytes           |
+| data block offset   | 8 bytes           |
++-----------------------------------------+
+```
+
+#### Account Meta Block
+
++--------------------------------------+
+| Hash (32 bytes for the entire block) |
++--------------------------------------+
+| account meta 1                       |
++--------------------------------------+
+| account meta 2                       |
++--------------------------------------+
+|         ...                          |
++--------------------------------------+
+| account meta N                       |
++--------------------------------------+
+
+```
++-------------------------------------------------------------+
+| Account Meta (16 bytes each)                                |
++-------------------------------------------------------------+
+| owner local id  | 4 bytes                                   |
+|                 | up to 4,294,967,295 owners per append vec |
+| data_offset     | 4 bytes                                   |
++-----------------+-------------------------------------------+
+| rent status     | 1 bit                                     |
+| executable      | 1 bit                                     |
+|                 | non-rent-exempt block will have its       |
+|                 | first 8-bytes of data to be rent-epoch    |
+| lamport         | 7 bytes                                   |
++-------------------------------------------------------------+
+```
+
+In memory, each StoredAccountMetaV2 occupies 24 bytes.  The extra
+8 bytes is to store the derived `local_id` for accessing the pubkey
+entry and the `data_size` for loading account data.
+
+```
+pub StoredAccountMetaV2 {
+    pub local_id: u32, // derived when loading the AppendVec from the storage
+                       // the id can be used to fetch its pubkey
+    pub owner_local_id: u32,
+    data_offset: u32,
+    data_size: u32,   // derived when first loading the AppendVec from the storage
```

Comment:
fwiw, data size is currently limited to 10MB, iirc.

Reply:
Do we happen to have some size unit for the data size?  Such as 1k or 64 bytes or anything like this?  If we have a size unit, then we can represent a wider range of data sizes with fewer bytes.